packages/@tinacms/graphql/src/builder/index.ts

```diff
@@ -920,21 +1095,74 @@ export class Builder {
     return this._buildTemplateData(collectableTemplate.template)
   }
 
+  private _connectionFilterBuilder = async ({
+    fieldName,
+    namespace,
+    collection,
+    collections,
+  }: {
+    fieldName: string
+    namespace: string[]
+    collection?: Collectable
+    collections?: Collectable[]
+  }) => {
+    let filter
+    if (collections) {
+      filter = astBuilder.InputValueDefinition({
+        name: 'filter',
+        type: astBuilder.InputObjectTypeDefinition({
+          name: NAMER.dataFilterTypeName(namespace),
+          fields: await sequential(collections, async (collection) => {
+            return astBuilder.InputValueDefinition({
+              // @ts-ignore
+              name: collection.name,
+              type: await this._filterCollectionDocumentType(collection),
+            })
+          }),
+        }),
+      })
+    } else if (collection) {
+      filter = astBuilder.InputValueDefinition({
+        name: 'filter',
+        type: await this._filterCollectionDocumentType(collection),
+      })
+    } else {
+      throw new Error(
+        `Must provide either collection or collections to filter field builder`
+      )
+    }
+    return filter
+  }
+
   private _connectionFieldBuilder = async ({
     fieldName,
     namespace,
     connectionName,
     nodeType,
+    collection,
+    collections,
   }: {
     fieldName: string
     namespace: string[]
     connectionName: string
     nodeType: string | TypeDefinitionNode
+    collection?: Collectable
+    collections?: Collectable[]
   }) => {
+    const extra = this.experimentalData
```

Comment:
The `filter` property will only appear when the `experimentalData` flag is enabled via the CLI.